dali/base/sysinfologger.cpp

```diff
@@ -746,23 +746,25 @@ class DaliMsgLoggerHandler : public CInterfaceOf<ILogMsgHandler>
     unsigned messageFields = MSGFIELD_all;
 };
 
+static Owned<ILogMsgHandler> msgHandler;
+
 void UseDaliForOperatorMessages(bool use)
 {
-    static ILogMsgHandler *msgHandler=NULL;
-    if (use==(msgHandler!=NULL))
-        return;
     if (use)
     {
-        msgHandler = getDaliMsgLoggerHandler();
-        ILogMsgFilter * operatorFilter = getCategoryLogMsgFilter(MSGAUD_operator,
-                                                                 MSGCLS_disaster|MSGCLS_error|MSGCLS_warning,
-                                                                 WarnMsgThreshold);
-        queryLogMsgManager()->addMonitorOwn(msgHandler, operatorFilter);
+        if (!msgHandler)
+        {
+            msgHandler.setown(getDaliMsgLoggerHandler());
+            ILogMsgFilter * operatorFilter = getCategoryLogMsgFilter(MSGAUD_operator,
```

Comment:
please check, but this looks like it leaks ? addMonitor (via LogMsgMonitor ctor) links both the handler and the filter.

Reply:
Regarding id's: I agree that different uses of the id may cause confusion and may lead to bugs in the future.  I have created a PR to distinguish between message id's and "sequence numbers".  Should this be merged first: https://github.com/hpcc-systems/HPCC-Platform/pull/19367 @jakesmith

Reply:
You're right.  It looks this will leak memory.   Similar code exists through-out jlog.cpp and it looks like addMonitorOwn may be leaking the handler and filter all over the place (e.g. attachStandardFileLogMsgMonitor, attachStandardBinLogMsgMonitor etc).

Reply:
Yes, it looks like a leak.